src/gui/shareusergroupwidget.cpp

```diff
@@ -231,8 +231,21 @@ void ShareUserGroupWidget::slotCompleterActivated(const QModelIndex & index)
         return;
     }
 
-    _manager->createShare(_sharePath, Share::ShareType(sharee->type()),
-                          sharee->shareWith(), Share::PermissionDefault);
+    /*
+     * Don't send the reshare permissions for federataed shares
+     * https://github.com/owncloud/core/issues/22122#issuecomment-185637344
+     */
+    if (sharee->type() == Sharee::Federated) {
```

Comment:
Yes, sure. It is crappy right now. We are working in the server to improve this. But for now this is the servers behaviour so we should somehom mirror this.